src/main/java/com/develhope/spring/user/model/User.java

```diff
@@ -0,0 +1,137 @@
+package com.develhope.spring.user.model;
+
+import com.develhope.spring.user.Account;
+
+public class User {
+
+    private String name;
+
+    private String surname;
+
+    private long mobile;
+
+    private String email;
+
+    private long password;
+
+    private Account account;
+
+    private static boolean isValidString(String value) {
+        return !value.isEmpty() && value.chars().allMatch(Character::isLetter);
+    }
+
+    private static boolean containsCommonPassword(long password) {
+        long[] COMMON_PASSWORDS = {123456L, 654321L, 9876543210L}; //esempi di pass comuni
+        for (long commonPassword : COMMON_PASSWORDS) {
+            if (password == commonPassword) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void checkValidName(String name) throws InvalidValueException {
+        String rightName = name.trim(); //rimuove i possibili spazi nella stringa
+        if (rightName.isEmpty() || !rightName.chars().allMatch(Character::isLetter)) {
+            throw new InvalidValueException("This field can't be empty and can't contains numbers or special character, please insert a valid input (Name)");
+        } else {
+            this.name = rightName;
+        }
+    }
+
+    private void checkValidSurname(String surname) throws InvalidValueException {
+        String rightSurname = surname.trim();
+        if (rightSurname.isEmpty() || !rightSurname.chars().allMatch(Character::isLetter)) {
+            throw new InvalidValueException("This field can't be empty and can't contains numbers or special character, please insert a valid input (Surname)");
+        } else {
+            this.name = rightSurname;
+        }
+    }
+
+    private void checkValidEmail(String email) throws InvalidEmailException, InvalidValueException {
+        if (email == null || email.trim().isEmpty()) {
+            throw new InvalidValueException("This field can't be empty, please insert a valid input (email)");
+        }
+        if (email.chars().filter(ch -> ch == '@').count() != 1 || !email.substring(email.indexOf('@')).contains(".")) { // Verifica se l'email contiene almeno una @ e se dopo questa ha almeno un .
+            throw new InvalidEmailException("invalid input, please insert right params");
+        }
+        this.email = email.trim();
+    }
+
+    private void checkValidPassword(long password) throws InvalidValueException {
```

Comment:
@Size(min=2, max=30): Allows names between 2 and 30 characters long.

Reply:
è necessario questo controllo secondo te?